smac/facade/smac_facade.py

```diff
@@ -184,19 +214,35 @@ def __init__(self,
             self.stats = Stats(scenario)
 
         if self.scenario.run_obj == "runtime" and not self.scenario.transform_y == "LOG":
-            self.logger.warn("Runtime as objective automatically activates log(y) transformation")
+            self.logger.warning("Runtime as objective automatically activates log(y) transformation")
             self.scenario.transform_y = "LOG"
 
         # initialize empty runhistory
+        runhistory_def_kwargs = {'aggregate_func': aggregate_func}
+        if runhistory_kwargs is not None:
+            runhistory_def_kwargs.update(runhistory_kwargs)
         if runhistory is None:
-            runhistory = RunHistory(aggregate_func=aggregate_func)
-        # inject aggr_func if necessary
-        if runhistory.aggregate_func is None:
-            runhistory.aggregate_func = aggregate_func
+            runhistory = RunHistory(**runhistory_def_kwargs)
+        elif inspect.isclass(runhistory):
+            runhistory = runhistory(**runhistory_def_kwargs)
+        else:
+            if runhistory.aggregate_func is None:
+                runhistory.aggregate_func = aggregate_func
 
-        if not random_configuration_chooser:
-            random_configuration_chooser = ChooserProb(prob=scenario.rand_prob,
-                                                       rng=rng)
+        rand_conf_chooser_kwargs = {
+           'rng': rng
+        }
+        if random_configuration_chooser_kwargs is not None:
+            rand_conf_chooser_kwargs.update(random_configuration_chooser_kwargs)
+        if random_configuration_chooser is None:
+            if 'prob' not in rand_conf_chooser_kwargs:
+                rand_conf_chooser_kwargs['prob'] = scenario.rand_prob
+            random_configuration_chooser = ChooserProb(**rand_conf_chooser_kwargs)
+        elif inspect.isclass(random_configuration_chooser):
+            random_configuration_chooser = random_configuration_chooser(**rand_conf_chooser_kwargs)
```

Comment:
Why is the injection of `prob` here missing, but done in line 239?

Reply:
`prob` is not a mandatory argument for a `RandomConfigurationChooser`, therefore, it is only passed to the chooser which requires it.

Reply:
but we don't know whether the class `random_configuration_chooser` will require it or not

Reply:
Correct. In fact, we know for no method except the default method what kind of hyperparameters it requires. To rephrase my statement, the abstract class `RandomConfigurationChooser` requires only the random state and therefore, only the random state is passed by default.

If you think that's not good we should rather add the default value to constructor of the respective class instead of adding it to this dictionary. Ideally, the dictionaries only add new "default" values which are constructed inside the smac facade. At some point in the future it would be good if we move all other arguments somewhere else.